third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -228,5 +229,35 @@ createConvertTritonGPUToLLVMPass(int32_t computeCapability) {
   return std::make_unique<ConvertTritonGPUToLLVM>(computeCapability);
 }
 
+bool NVIDIA::canSkipBarSync(Operation *before, Operation *after) {
+  if (isa<triton::nvidia_gpu::InitBarrierOp>(before) &&
+      isa<triton::nvidia_gpu::InitBarrierOp>(after))
+    return true;
+
+  if (isa<triton::nvidia_gpu::InvalBarrierOp>(before) &&
+      isa<triton::nvidia_gpu::InvalBarrierOp>(after))
+    return true;
```

Comment:
If we confirm that `init_barrier` or `inval_barrier` followed with an immediate `init_barrier` or `inval_barrier` on the same operand, we can remove the above conditions

Reply:
yeah, I explained why I did it in the other comment